lib/internal/Magento/Framework/Module/ModuleList/Loader.php

```diff
@@ -97,7 +105,7 @@ public function load(array $exclude = [])
                 $result[$name] = $data[$name];
             }
         }
-        return $this->sortBySequence($result);
+        return $this->sorter->sort($result);
```

Comment:
Something like `SequenceSorter` seems to be a more appropriate name (both class and injected argument).